tests/unit/test_credentials.py

```diff
@@ -972,5 +973,137 @@ def test_source_profile_does_not_exist(self):
             provider.load()
 
 
+class TestSamlGenericFormsBasedAuthenticator(unittest.TestCase):
+    authenticator = credentials.SamlGenericFormsBasedAuthenticator()
```

Comment:
these should be added to setUp().  That's the standard way of test fixture setup.